chia/data_layer/data_layer_server.py

```diff
@@ -58,3 +102,41 @@ async def file_handler(self, request: web.Request) -> web.Response:
             body=content,
         )
         return response
+
+    def _accept_signal(self, signal_number: int, stack_frame: Any = None) -> None:
+        self.log.info("Got SIGINT or SIGTERM signal - stopping")
+
+        self.stop()
+
+
+async def async_start(config: Optional[str]) -> int:
+
+    shutdown_event = asyncio.Event()
+
+    if config:
+        root_path = Path(config)
+    else:
+        root_path = DEFAULT_ROOT_PATH
+
+    dl_config = load_config(root_path=root_path, filename="config.yaml", sub_config=SERVICE_NAME)
+    setproctitle("data_layer_http")
+    initialize_logging(
+        service_name="data_layer_http",
+        logging_config=dl_config["logging"],
+        root_path=root_path,
+    )
+
+    data_layer_server = DataLayerServer(root_path, dl_config, log, shutdown_event)
+    await data_layer_server.start()
+
+    return 0
+
+
+@click.command()
+@click.option("-c", "--config", type=click.Path(exists=True, writable=True, dir_okay=True), default=None)
+def main(config: Optional[str] = None) -> int:
```

Comment:
It does seem odd to convert a Path (DEFAULT_ROOT_PATH) into a string only to convert it back to a Path for load_config - however, I have made this change.

Reply:
Mm, I meant that the function parameter shouldn't have a default at all.  That's not normally done with click since click will either provide a value when calling the function or just not even call the function if a required option is not provided on the cli.

As to the `str` type used for `click.Path`, yeah...  I believe I submitted a PR to make it optionally provide a `Path` back but they didn't like it and later wrote some other approach up.  I think that's in 8.something though.  In the mean time, for the default specifically, I guess as shown by the other code the click default parameter works out ok as just a `Path`.  Though it probably is lying a bit with the hinting.

Reply:
Without a default, mypy complains about the `sys.exit(main())` call, which I would imagine I would have to call with the default `sys.exit(main(str(DEFAULT_ROOT_PATH))` - which doesn't seem much better